python-avd/pyavd/_eos_designs/shared_utils/wan.py

```diff
@@ -170,9 +212,15 @@ def wan_ha_peer_path_group_names(self: SharedUtils) -> list:
         """Return a list of wan_ha_peer_path_group names."""
         return [path_group["name"] for path_group in self.wan_ha_peer_path_groups]
 
-    def get_public_ip_for_wan_interface(self: SharedUtils, interface: EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3InterfacesItem) -> str:
+    def get_public_ip_for_wan_interface(
+        self: SharedUtils,
+        interface: (
+            EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3InterfacesItem
+            | EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannelsItem
+        ),
+    ) -> str:
         """
-        Takes a dict which looks like `l3_interface` from node config.
+        Takes a dict which looks like `l3_interface` or `l3_port_channel` from node config.
```

Comment:
well it is not a dict anymore. It is the class. So IMO we can remove this line.
```suggestion
```

Reply:
Ack